example/src/test/java/com/oviva/spicegen/example/ExampleTest.java

```diff
@@ -109,10 +118,18 @@ void example() {
             document.checkRead(
                 SubjectRef.ofObject(user), Consistency.atLeastAsFreshAs(consistencyToken))));
 
-    assertTrue(
-        permissionService.checkPermission(
-            document.checkRead(
-                SubjectRef.ofObject(user2), Consistency.atLeastAsFreshAs(consistencyToken))));
+    Iterator<UserRef> usersAllowedToRead =
+        permissionService.lookupSubjects(document.lookupReadUser());
+    assertTrue(usersAllowedToRead.hasNext());
+    // usersAllowedToRead contains both userId and user2
+    var userIds =
+        StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(usersAllowedToRead, Spliterator.ORDERED), false)
+            .map(UserRef::id)
+            .toList();
+    assertEquals(2, userIds.size());
+    assertTrue(userIds.contains(user.id()));
```

Comment:
while writing these assertions i noticed that implementations of ObjectRef do not implement equals/hashCode based on the id.
I think they should + they could be records.
looks like Square archived the original javapoet, but there are forks that support records i.e: https://github.com/palantir/javapoet